### ISSUE TITLE
Split up partials into partial events

### ISSUE THREAD

**Issue by Jiralite** [discordjs/discord.js#7045]
### Feature

I feel a good addition (or refactor) to discord.js would be to use partial events.

Currently, when we have a partial message update, we are handling everything within the `messageUpdate` event. The parameters transform from the typings to also handle the partial messages. I feel it would be a nice change to separate these out and emit partial events for partial structures and leave full events for full structures, which would seem to make the end user's code handling better.

### Ideal solution or implementation

For those who don't utilise partials, there would be no change. For those who utilise partials, it would look something like this:

```JavaScript
const { Client, Constants, Intents } = require("discord.js");
const { functionThatRequiresFullUpdatedMessage } = require("./MyAPI/narnia.js");

const client = new Client({
  partials: [
    Constants.PartialTypes.MESSAGE
  ],
  intents: [
    Intents.FLAGS.GUILDS,
    Intents.FLAGS.GUILD_MESSAGES
  ]
});

client.on(Constants.Events.MESSAGE_UPDATE, (oldMessage, newMessage) => {
  // Only full discord.js structures are emitted here. We may safely proceed without handling partials!
  functionThatRequiresFullUpdatedMessage(newMessage);
});

client.on(Constants.Events.PARTIAL_MESSAGE_UPDATE /* soon:tm: */, (oldPartialMessage, newPartialMessage) => {
  // Only partial discord.js structures are emitted here. We should handle them accordingly.
  const newMessage = newPartialMessage.fetch();
  functionThatRequiresFullUpdatedMessage(newMessage);
});
```

I was also thinking of partial structures. It seems we already use them in the typings but they're not documented. Instead of using the full structure for partial events, what about initialising partial structures instead with only the received data? This could look clearer in the documentation: the full structures will no longer have questionable nullable properties because of partials and only the relevant partial structure documented will possess them.

**Comment by ImRodry** [comment]
I agree with the documentation changes but splitting the same kind of event (and likely the same code) into two handlers seems unnecessary to me. You just gotta put the checks in place to account for partial data. Forcing users to duplicate their code doesn’t seem like a good idea to me
